bundles/org.openhab.core.persistence/src/test/java/org/openhab/core/persistence/internal/PersistenceManagerTest.java

```diff
@@ -235,10 +237,52 @@ public void appliesToItemWithAllConfig() {
         verifyNoMoreInteractions(persistenceServiceMock);
     }
 
+    @Test
+    public void doesNotApplyToItemWithGroupConfigAndItemExclusion() {
+        addConfiguration(TEST_PERSISTENCE_SERVICE_ID, List.of(new PersistenceGroupConfig(TEST_GROUP_ITEM_NAME),
+                new PersistenceItemExcludeConfig(TEST_ITEM_NAME)), PersistenceStrategy.Globals.UPDATE, null);
+
+        manager.stateUpdated(TEST_ITEM, TEST_STATE);
+
+        verifyNoMoreInteractions(persistenceServiceMock);
+    }
+
+    @Test
+    public void doesNotApplyToItemWithAllConfigAndItemExclusion() {
+        addConfiguration(TEST_PERSISTENCE_SERVICE_ID,
+                List.of(new PersistenceAllConfig(), new PersistenceItemExcludeConfig(TEST_ITEM_NAME)),
+                PersistenceStrategy.Globals.UPDATE, null);
+
+        manager.stateUpdated(TEST_ITEM, TEST_STATE);
+        manager.stateUpdated(TEST_ITEM2, TEST_STATE);
+        manager.stateUpdated(TEST_GROUP_ITEM, TEST_STATE);
+
+        verify(persistenceServiceMock).store(TEST_ITEM2, null);
+        verify(persistenceServiceMock).store(TEST_GROUP_ITEM, null);
+
+        verifyNoMoreInteractions(persistenceServiceMock);
+    }
+
+    @Test
+    public void doesNotApplyToItemWithAllConfigAndGroupExclusion() {
+        addConfiguration(TEST_PERSISTENCE_SERVICE_ID,
+                List.of(new PersistenceAllConfig(), new PersistenceGroupExcludeConfig(TEST_GROUP_ITEM_NAME)),
+                PersistenceStrategy.Globals.UPDATE, null);
+
+        manager.stateUpdated(TEST_ITEM, TEST_STATE);
+        manager.stateUpdated(TEST_ITEM2, TEST_STATE);
+        manager.stateUpdated(TEST_GROUP_ITEM, TEST_STATE);
+
+        verify(persistenceServiceMock).store(TEST_ITEM2, null);
+        verify(persistenceServiceMock).store(TEST_GROUP_ITEM, null);
```

Comment:
Not sure if I get this test correctly. Could you pls. explain why the group item is here, and Item1 is excluded?

Reply:
States of TEST_ITEM, TEST_ITEM2 and TEST_GROUP_ITEM are updated. The persistence configuration is to persist all except items inside TEST_GROUP_ITEM. TEST_ITEM is in TEST_GROUP_ITEM.
Therefore, the right behaviour is to persist TEST_ITEM2 and TEST_GROUP_ITEM itself (not its members).

Reply:
> TEST_ITEM is in TEST_GROUP_ITEM. 

This is exactly what I was looking for in the code. Got it now. Don't know why I overlooked line 143.

Thanks.